packages/bot-skeleton/src/services/api/api-base.js

```diff
@@ -55,25 +56,41 @@ class APIBase {
         }
     };
 
-    authorizeAndSubscribe() {
+    async authorizeAndSubscribe() {
         const { token, account_id } = getToken();
         if (token) {
             this.token = token;
             this.account_id = account_id;
-            this.api
-                .authorize(this.token)
-                .then(({ authorize }) => {
-                    if (this.has_activeSymbols) {
-                        this.toggleRunButton(false);
-                    } else {
-                        this.getActiveSymbols();
-                    }
-                    this.subscribe();
-                    this.account_info = authorize;
-                })
-                .catch(e => {
-                    globalObserver.emit('Error', e);
-                });
+            try {
+                const auth_response = await this.api.authorize(this.token);
+                if (auth_response.error) {
+                    throw new Error(auth_response.error);
+                }
+                if (this.has_activeSymbols) {
+                    this.toggleRunButton(false);
+                } else {
+                    this.getActiveSymbols();
+                }
+                await WS.wait('authorize');
+                this.subscribe();
+            } catch (e) {
+                globalObserver.emit('Error', e);
+            }
+            // this.api
```

Comment:
please remove the commented code if its not needed any more

Reply:
Sure.I will remove it